test/hotspot/jtreg/serviceability/jvmti/vthread/InterruptRawMonitor/InterruptRawMonitor.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test id=default
+ * @bug 8325187
+ * @summary Verifies JVMTI InterruptThread works for virtual threads.
+ * @run main/othervm/native -agentlib:InterruptRawMonitor InterruptRawMonitor
+ *
+ * @test id=virtual
+ * @bug 8325187
+ * @summary Verifies JVMTI InterruptThread works for virtual threads.
+ * @run main/othervm/native -agentlib:InterruptRawMonitor InterruptRawMonitor -v
+ */
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class InterruptRawMonitor {
+    private static final String AGENT_LIB = "InterruptRawMonitor";
+    static native void test();
+
+    public static void main(String[] args) throws Exception {
+        Thread thread;
+        if (args.length > 0 && "-v".equals(args[0])) {
+            thread = Thread.ofVirtual().unstarted(InterruptRawMonitor::test);
+        } else {
+            thread = Thread.ofPlatform().unstarted(InterruptRawMonitor::test);
+        }
+        System.out.println(thread);
+        thread.start();
+        Thread.sleep(2000);
```

Comment:
I think it would be better to check 'thread' status or the same monitor  to sync instead of sleep.  The sleep is always looks suspect, especially when intermittent failure happens. 
Also, it helps to save 2 seconds.

Reply:
Will implement this suggestion, thanks.